src/OpenTelemetry.Instrumentation.SqlClient/Implementation/SqlClientDiagnosticListener.cs

```diff
@@ -210,6 +212,13 @@ public override void OnEventWritten(string name, object? payload)
                         {
                             if (this.exceptionFetcher.TryFetch(payload, out Exception? exception) && exception != null)
                             {
+                                activity.AddTag(SemanticConventions.AttributeErrorType, exception.GetType().FullName);
+
+                                if (this.exceptionNumberFetcher.TryFetch(exception, out var exceptionNumber))
+                                {
+                                    activity.AddTag(SemanticConventions.AttributeDbResponseStatusCode, exceptionNumber.ToString(CultureInfo.InvariantCulture));
+                                }
+
                                 activity.SetStatus(ActivityStatusCode.Error, exception.Message);
 
                                 if (this.options.RecordException)
```

Comment:
The instrumentation is currently setting the span's status to ERROR when an exception occurs. A span event is also recorded with the exception if the user opts into this.

I do not think the conventions currently specify setting span status or recording the exception event. Should they?

Reply:
> The instrumentation is currently setting the span's status to ERROR when an exception occurs. A span event is also recorded with the exception if the user opts into this.

this seems great for exceptions that are thrown to the user. 

> I do not think the conventions currently specify setting span status or recording the exception event. Should they?

Great point. Let me create an issue.